docs/content/v2025.1/explore/transactions/transactional-ddl.md

```diff
@@ -0,0 +1,134 @@
+
+Because `yb_ddl_transaction_block_enabled` is a preview flag, to use it, add the flag to the [allowed_preview_flags_csv](../../../reference/configuration/yb-tserver/#allowed-preview-flags-csv) list (that is, `allowed_preview_flags_csv=yb_ddl_transaction_block_enabled`).
+
+### Rollback capabilities
+
+All DDLs supported in YugabyteDB provide the same rollback capabilities as PostgreSQL. These include DDLs on tables, indexes, roles, and materialized views.
+
+Note that some DDL statements such as DDLs on database or tablespaces are disallowed in a transaction block in PostgreSQL, and are also disallowed in YugabyteDB.
+
+#### Rollback CREATE TABLE
+
+This example demonstrates rolling back a `CREATE TABLE` statement within a transaction block.
+
+```sql
+yugabyte=# BEGIN;
+yugabyte=*# CREATE TABLE foo (bar int);
+yugabyte=*# INSERT INTO foo VALUES (1);
+yugabyte=*# ROLLBACK;
+yugabyte=# SELECT * FROM foo;
+```
+
+```output
+ERROR:  relation "foo" does not exist
+LINE 1: SELECT * FROM foo;
+                      ^
+```
+
+#### Rollback ALTER TABLE
+
+This example demonstrates rolling back an `ALTER TABLE` statement that adds a column.
+
+```sql
+yugabyte=# CREATE TABLE foo (bar int);
+yugabyte=# BEGIN;
+yugabyte=*# ALTER TABLE foo ADD COLUMN name text;
+yugabyte=*# SELECT * FROM foo;
+```
+
+```output
+ bar | name
+-----+------
+(0 rows)
+```
+
+```sql
+yugabyte=*# ROLLBACK;
+yugabyte=# SELECT * FROM foo;
+ bar
+-----
+(0 rows)
+```
+
+#### Rollback Materialized View
+
+This example demonstrates rolling back the creation of a materialized view.
+
+```sql
+yugabyte=# CREATE TABLE t4 (id int);
+yugabyte=# INSERT INTO t4 VALUES (1);
+yugabyte=# BEGIN;
+yugabyte=*# CREATE MATERIALIZED VIEW mv4 AS SELECT * FROM t4;
+yugabyte=*# ROLLBACK;
+yugabyte=# SELECT * FROM mv4;
+```
+
+```output
+ERROR:  relation "mv4" does not exist
+LINE 1: SELECT * FROM mv4;
+                      ^
+```
+
+#### Rollback ALTER ROLE
+
+This example demonstrates rolling back an `ALTER ROLE` statement that grants a privilege.
+
+```sql
+yugabyte=# CREATE ROLE test_user;
+yugabyte=#
+yugabyte=# BEGIN;
+yugabyte=*# ALTER ROLE test_user WITH CREATEDB;
+yugabyte=*# \du test_user
+```
+
+```output
+                  List of roles
+ Role name |       Attributes        | Member of
+-----------+-------------------------+-----------
+ test_user | Create DB, Cannot login | {}
+```
+
+```sql
+yugabyte=*# ROLLBACK;
+
+yugabyte=# -- Role doesn't have CREATEDB privilege
+yugabyte=# \du test_user
+```
+
+```output
+            List of roles
+ Role name |  Attributes  | Member of
+-----------+--------------+-----------
+ test_user | Cannot login | {}
+```
+
+### Limitations
+
+- Concurrent DDLs on the same database are unsupported and will lead to conflict and read restart required errors. Your applications must handle these by retrying the statements.
```

Comment:
This should link to https://docs.yugabyte.com/preview/best-practices-operations/administration/#concurrent-ddl-during-a-ddl-operation where we explain this in more detail.